apps/server/lib/lexical/server/transport/std_io.ex

```diff
@@ -76,27 +83,33 @@ defmodule Lexical.Server.Transport.StdIO do
     end
   end
 
-  defp parse_headers(headers) do
-    Enum.map(headers, &parse_header/1)
-  end
+  defp content_length(headers) do
+    case List.keyfind(headers, "content-length", 0) do
+      {_, len_str} ->
+        {:ok, String.to_integer(len_str)}
 
-  defp header_value(headers, header_name, converter) do
-    case List.keyfind(headers, header_name, 0) do
-      nil -> :error
-      {_, value} -> {:ok, converter.(value)}
+      nil ->
+        {:error, :no_content_length_header}
     end
   end
 
   defp read_body(device, byte_count) do
     case IO.binread(device, byte_count) do
-      data when is_binary(data) or is_list(data) ->
+      data when is_binary(data) ->
         {:ok, data}
 
-      other ->
-        other
+      :eof ->
+        {:error, :eof}
```

Comment:
I'm pretty sure we want to shut down in this case, as it signals that the stream is closed.

Reply:
Good point, think this would work?
```elixir
:eof ->
  Logger.critical("stdio received :eof, server will stop.")
  maybe_stop()
```

Reply:
Sure, though it's likely you won't see that message.

Reply:
Added